src/lib/commands.js

```diff
@@ -0,0 +1,271 @@
+
+		getVendorList((vendorList, err) => {
+			if (err) {
+				callback({}, false);
+			}
+			if (vendorList) {
+				const {
+					persistedConsentData,
+				} = store;
+
+				const {
+					created,
+					lastUpdated,
+					consentLanguage_: consentLanguage,
+					consentScreen_: consentScreen,
+					version_: version = 1,
+					cmpId_: cmpId = 280
+				} = persistedConsentData;
+
+				const {
+					selectedVendorIds = [],
+					selectedPurposeIds = [],
+					maxVendorId,
+					cookieVersion,
+					vendorListVersion,
+					cmpVersion
+				} = data;
+
+				const consentData = {
+					cmpId,
+					maxVendorId,
+					cookieVersion,
+					cmpVersion,
+					vendorListVersion,
+					created,
+					lastUpdated,
+					consentLanguage,
+					consentScreen,
+					version,
+					selectedVendorIds: new Set(arrayFrom(selectedVendorIds)),
+					selectedPurposeIds: new Set(arrayFrom(selectedPurposeIds))
+				};
+
+				const valid = (data = {}) => [
+					'cmpId', 'cmpVersion', 'consentLanguage', 'consentScreen', 'cookieVersion', 'created', 'lastUpdated',
+					'maxVendorId', 'selectedPurposeIds', 'selectedVendorIds', 'vendorListVersion'
+				].every(prop => data.hasOwnProperty(prop));
+
+				callback(valid(consentData) ? encodeVendorConsentData(consentData) : undefined);
+			}});
+	};
+
+	const getConsentFieldsV1 = (callback, params) => {
+		const data = getConsentFieldsObject();
+		const now = new Date();
+
+		generateConsentStringV1({
+			created: now,
+			lastUpdated: now,
+			...data,
+			...params
+		}, (metadata) => {
+			const consent = {
+				metadata,
+				gdprApplies: config.gdprApplies,
+				hasGlobalScope: config.storeConsentGlobally,
+				...data,
+				vendorListVersion: params.vendorListVersion
+			};
+
+			callback(consent, true);
+		});
+	};
+
+	const getConsentObject = (callback, params) => {
+		getVendorList((vendorList, err) => {
+			if (err) {
+				callback({}, false);
+			}
+			if (vendorList) {
+				const {
+					vendorListVersion: paramsVendorListVersion,
+					...tcModelFields
+				} = params;
+				const {persistedConsentData} = store;
+				const {vendors} = vendorList;
+				const globalVendorsObject = {};
+
+				for (let vendor of Object.values(vendors)) {
+					if (!vendor.globalId) {
+						continue;
+					}
+
+					globalVendorsObject[vendor.globalId] = {
+						id: vendor.globalId,
+						name: vendor.name,
+						policyUrl: vendor.policyUrl,
+						purposes: vendor.purposes,
+						specialPurposes: vendor.specialPurposes,
+						legIntPurposes: vendor.legIntPurposes,
+						features: vendor.features,
+						specialFeatures: vendor.specialFeatures
+					};
+				}
+
+				vendorList.vendors = globalVendorsObject;
+				vendorList.vendorListVersion = paramsVendorListVersion;
+
+				const tcModel = new TCModel(new GVL(vendorList));
+				tcModel.cookieVersion = COOKIE_VERSION;
+				tcModel.cmpId = CMP_ID;
+				tcModel.cmpVersion = CMP_VERSION;
+				tcModel.created = persistedConsentData && persistedConsentData.created || new Date();
```

Comment:
If persisted data doesn't exist we should set `undefined` instead of the current time here. Same case below